packages/gator-permissions-snap/src/permissions/erc20TokenStream/types.ts

```diff
@@ -28,8 +28,8 @@ export type Erc20TokenStreamMetadata = BaseMetadata & {
 
 export type Erc20TokenStreamContext = BaseContext & {
   permissionDetails: {
-    initialAmount: string | undefined;
-    maxAmount: string | undefined;
+    initialAmount: string | undefined | null;
```

Comment:
nit: Because the context is always constructed, could we define this more tightly as `string | null`?

I don't think it matters a _lot_ especially in the light of an upcoming change, where we likely won't actually set it to undefined / null, rather have a flag indicating that it is removed, but it could tidy things up a bit 🤷 

Same comment, obviously for the other permission types.

Reply:
We have zod validation to allow optional, so accepting undefined. This would also mean changing that and everything that comes with it. If that is what we want then I can update everything. But that could potentially be a breaking change?